File: JavaScript/프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    var answer = 0;
    var temp, s_temp, idx, i;
    for (let ex of skill_trees) {
        temp = [];
        for (let str of skill) {
            i = ex.search(str);
            idx = (i == -1) ? Infinity : i;
            temp.push(idx);
        }
        s_temp = [...temp].sort((a,b) => a - b);
        if (JSON.stringify(temp) == JSON.stringify(s_temp)) answer++;
    }
    
    return answer;
}

File: Python3/프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    
    for ex in skill_trees:
        temp = []
        for str in skill:
            idx = float('inf')
            if str in ex:
                idx = ex.find(str)
            temp.append(idx)
        print(f"temp:{temp}")
        print(f"sort temp:{sorted(temp)}\n")
        if temp == sorted(temp):
            answer += 1
    return answer